src/plugins/links/public/components/dashboard_link/dashboard_link_component.tsx

```diff
@@ -0,0 +1,171 @@
+}) => {
+  const linksEmbeddable = useLinks();
+  const [error, setError] = useState<Error | undefined>();
+
+  const dashboardContainer = linksEmbeddable.parent as DashboardContainer;
+  const parentDashboardInput = useObservable(dashboardContainer.getInput$());
+  const parentDashboardId = dashboardContainer.select((state) => state.componentState.lastSavedId);
+
+  /** Fetch the dashboard that the link is pointing to */
+  const { loading: loadingDestinationDashboard, value: destinationDashboard } =
+    useAsync(async () => {
+      if (link.id !== parentDashboardId && link.destination) {
+        /**
+         * only fetch the dashboard if it's not the current dashboard - if it is the current dashboard,
+         * use `dashboardContainer` and its corresponding state (title, description, etc.) instead.
+         */
+        const dashboard = await fetchDashboard(link.destination)
+          .then((result) => {
+            setError(undefined);
+            return result;
+          })
+          .catch((e) => setError(e));
+        return dashboard;
+      }
+    }, [link, parentDashboardId]);
+
+  /**
+   * Returns the title and description of the dashboard that the link points to; note that, if the link points to
+   * the current dashboard, then we need to get the most up-to-date information via the `parentDashboardInput` - this
+   * will respond to changes so that the link label/tooltip remains in sync with the dashboard title/description.
+   */
+  const [dashboardTitle, dashboardDescription] = useMemo(() => {
+    return link.destination === parentDashboardId
+      ? [parentDashboardInput?.title, parentDashboardInput?.description]
+      : [destinationDashboard?.attributes.title, destinationDashboard?.attributes.description];
+  }, [link.destination, parentDashboardId, parentDashboardInput, destinationDashboard]);
+
+  /**
+   * Memoized link information
+   */
+  const linkLabel = useMemo(() => {
+    return link.label || (dashboardTitle ?? DashboardLinkStrings.getDashboardErrorLabel());
+  }, [link, dashboardTitle]);
+
+  const { tooltipTitle, tooltipMessage } = useMemo(() => {
+    if (error) {
+      return {
+        tooltipTitle: DashboardLinkStrings.getDashboardErrorLabel(),
+        tooltipMessage: error.message,
+      };
+    }
+    return {
+      tooltipTitle: Boolean(dashboardDescription) ? linkLabel : undefined,
+      tooltipMessage: dashboardDescription || linkLabel,
+    };
+  }, [error, linkLabel, dashboardDescription]);
+
+  /**
+   * Dashboard-to-dashboard navigation
+   */
+  const { loading: loadingOnClickProps, value: onClickProps } = useAsync(async () => {
+    /** If the link points to the current dashboard, then there should be no `onClick` or `href` prop */
+    if (link.destination === parentDashboardId) return;
+
+    const linkOptions = {
+      ...DEFAULT_DASHBOARD_DRILLDOWN_OPTIONS,
+      ...link.options,
+    } as DashboardDrilldownOptions;
+
+    const locator = await getDashboardLocator({
```

Comment:
This won't actually be necessary once https://github.com/elastic/kibana/issues/164748 is in, since it will be stored in the parent/container and we'll just be able to grab it without any sort of `await` - just wish it could have made it in to this 😅